hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -503,6 +501,15 @@ private void doBucketProbing() throws IOException {
     }
   }
 
+  private void logDnsLookup() {
+    String endPoint = getConf().getTrimmed(ENDPOINT, "");
```

Comment:
* make the default "s3.amazonaws.com" so we fall back to that?
* AWS docs imply that this can be a full https:// URL; never tested that to see what happens in our own code, but we may want to handle it here.

Reply:
> * make the default "s3.amazonaws.com" so we fall back to that?

Yes we can fall back as s3 client uses this by default if endpoint is not set explicitly. 

> * AWS docs imply that this can be a full https:// URL; never tested that to see what happens in our own code, but we may want to handle it here.

For this we have to trim the prefix. Also what if a port is present in the endpoint like https://<host>:port. Should I trim the port as well?
There is no method to do this out of box in NetUtils but I can do that.